controller/healthsyncer/healthsyncer.go

```diff
@@ -101,12 +103,15 @@ func updateConditionStatus(cluster *v3.Cluster, conditionType v3.ClusterConditio
 			condition.LastTransitionTime = currTime
 		}
 		condition.LastUpdateTime = currTime
+		condition.Reason = msg
 		cluster.Status.Conditions[pos] = *condition
 	} else {
 		ncondition := &v3.ClusterCondition{
```

Comment:
The condition should be added with status='unknown' on cluster create. Your handler should just update it